### PR TITLE
Added new guides to gen4, gen5 and universal

### PR DESCRIPTION
**Pull request by SpagoAsparago**
# Name of Resource
hex_editing.md
pt-hgss_reusabletms.md
bw_b2w2-buildings.md
bw_b2w2-map_insertion.md

# Description
Guides for hex editing (Universal), reusable tms in gen 4 games (gen4/field), map and buildings (gen5/field)
Also created the gen5/guides/field and its resources folders in gen5/guides, structured like the one in gen4

# Which Generation (4/5/Universal)?

# Notes
Also fixed an error in pt_hgss-pokemarts.md

### CODE REVIEW

**Review by PlatinumMaster — APPROVED** [review]

File: src/gen5/guides/field/bw_b2w2-map_insertion.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+3) If you map is using textures not present in the original map, then the *Area* value as well
+
+![](resources/bw_b2w2-map_insertion/mapinsertion.png)
+
+## Finding the map file
+Now you have to find the file number of the map you want to replace, you can do so in two ways:
+
+* Using SDSME:
+
+**SDSME is very old and not recommended, it's recommended to only use it for viewing purposes and don't make any edits with it**.
```
Comment (Gonhex) [inline src/gen5/guides/field/bw_b2w2-map_insertion.md:31]: This sentence needs to be revised. The content is good but difficult to understand on first reading.

**Review by Gonhex — COMMENTED** [review]
In this review I only looked over `bw_b2w2-map_insertion.md`. Overall it looks good but there are some points where I would request changes, if it wasn't already merged. At some points I wasn't sure about comma placements and capitalization. Also, for someone new to this, it might be easier to follow if the step-by-step guides had screenshots visualizing what to do. Can hide them in drop-downs to avoid bloating the site.

**Comment by PlatinumMaster** [comment]
Thanks for the input. We could revise it; I just merged them to get it on the site so there was *some* basic thing we could revise later.

File: src/gen5/guides/field/bw_b2w2-map_insertion.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Inserting your own Maps in Gen V games
+> This guide was written by Gonhex and was formatted and expanded for this wiki by SpagoAsparago
```
[the remark this answers is not in the document]
Comment (SpagoAsparago) [reply src/gen5/guides/field/bw_b2w2-map_insertion.md:2]: It's based on your pinned post in g5-mapping channel, I changed to information provided

File: src/gen5/guides/field/bw_b2w2-map_insertion.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+## Finding the map file
+Now you have to find the file number of the map you want to replace, you can do so in two ways:
+
+* Using SDSME:
+
+**SDSME is very old and not recommended, it's recommended to only use it for viewing purposes and don't make any edits with it**.
+
+Go to the Matrix Editor tab and select your Matrix. Then in the Map Headers tab find out where your Header number is located. Open the Map Files tab, and the number you're looking for will be at the same spot of that header in the previous tab. If there is no Map Header tab for the Matrix you've selected, it means all Map Files share the same header.
+
+* Using GFMatrixEditor:
```
[the remark this answers is not in the document]
Comment (SpagoAsparago) [reply src/gen5/guides/field/bw_b2w2-map_insertion.md:35]: At the moment I put the Google Drive link

**Comment by SpagoAsparago** [comment]
I fixed the typos and revised all listed changes: https://github.com/SpagoAsparago/kingdom-of-ds-hacking.github.io/blob/main/src/gen5/guides/field/bw_b2w2-map_insertion.md, what other screenshot do you think would help?

**Comment by Gonhex** [comment]
Nice. Forget about the screenshots for now, it was just a random idea I had but looking at it again it doesn't seem necessary.

**Pull request merged**